crypto/cipher/nacl/decrypter.go

```diff
@@ -24,19 +23,14 @@ type Decrypter struct {
 
 // Decrypt data using recipient private key with AES in CBC mode.
 func (d Decrypter) Decrypt(data cipher.EncryptedContent) (cipher.PlainContent, error) {
-	privKeyBytes, err := validatePrivateKeyType(d.privateKey)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-
-	return easyOpen(data, privKeyBytes)
+	return easyOpen(data, d.privateKey.Bytes()[32:])
```

Comment:
Different keys may have different ranges for returning the bytes to use when decrypting. If `[32:]` is done here you have then it is not aware of what type of key it is and it may pick the incorrect bytes.